src/environment.F

```diff
@@ -600,6 +602,8 @@ SUBROUTINE read_global_section(root_section, para_env, globenv)
          CALL cite_reference(Marek2014)
       CASE (FM_DIAG_TYPE_CUSOLVER)
          globenv%diag_library = "cuSOLVER"
+      CASE (FM_DIAG_TYPE_DLAF)
```

Comment:
ifdef statement needed here. CP2K should exit right away if dlaf support is turned off. The message should indicate that DLAF support is OFF explicitly.